src/Data/Decimal.purs

```diff
@@ -0,0 +1,190 @@
+module Data.Decimal
+  ( Decimal()
+  , decimal
+  , getMantissa
+  , getExponent
+  , fromInt
+  , prettyDecimal
+  ) where
+
+import Prelude
+import Data.Array as A
+import Data.Either as E
+import Data.Maybe as M
+import Data.Ord as O
+import Data.String as S
+import Control.Monad.Rec.Class as Rec
+
+import Test.StrongCheck as SC
+import Test.StrongCheck.Gen as Gen
+
+type DecimalR =
+  { mantissa :: Int
+  , exponent :: Int
+  }
+
+-- | An abstract type for exact computing with decimal numbers of finite expansion.
+newtype Decimal = Decimal DecimalR
+
+instance showDecimal :: Show Decimal where
+  show (Decimal d) =
```

Comment:
@garyb hmm; since the type is abstract, the only ways to produce it are using `decimal`, `fromInt` or any of the `Ring` operations. what do you suggest?

Maybe just a readable notation like `-123e-56`?